File: gcc/testsuite/gfortran.dg/enum_3.f90
! { dg-do compile }
! Program to test ENUM parsing errors 

program main
  implicit none
  enum, bind (c)
    enumerator :: red, black = 2.2  ! { dg-error "initialized with integer expression" }
    enumerator :: blue = "x"  ! { dg-error "initialized with integer expression" }
  end enum  ! { dg-error "has no ENUMERATORS" }

end program main

File: gcc/testsuite/gcc.dg/enum3.c
/* Test for non-integer enum values.  */
/* Origin: Joseph Myers <joseph@codesourcery.com> */
/* { dg-do compile } */
/* { dg-options "" } */

enum e { E, F };
enum e2 {
  E1 = (void *)4, /* { dg-error "enumerator value for 'E1' is not an integer constant" } */
  E2 = (enum e)F,
  E3 = (_Bool)1
};
